### PR TITLE
Chore/refactor assetserver

### PR DESCRIPTION
**Pull request by leaanthony**
Refactors the asset server with the following changes:

- removed ctx
- removed logger dep
- removed runtime dep
- removed fs dep

### CODE REVIEW

**Review by stffabi — APPROVED** [review]
LGTM 👍

File: v2/internal/frontend/desktop/darwin/frontend.go [hunk 1]
```diff
@@ -86,7 +87,8 @@ func NewFrontend(ctx context.Context, appoptions *options.App, myLogger *logger.
 		} else {
 			appBindings.DB().UpdateObfuscatedCallMap()
 		}
-		assets, err := assetserver.NewAssetServerMainPage(ctx, bindings, appoptions)
+
+		assets, err := assetserver.NewAssetServerMainPage(bindings, appoptions, ctx.Value("assetdir") != nil, myLogger, runtime.RuntimeAssetsBundle)
```
Comment (stffabi) [inline v2/internal/frontend/desktop/darwin/frontend.go:91]: Linux and Windows also contain this call in their frontend.go. Those changes are currently missing in the PR.

Comment (leaanthony) [reply v2/internal/frontend/desktop/darwin/frontend.go:91]: Ah the perils of build tags and IDEs 😭

**Comment by leaanthony** [comment]
@stffabi - thanks for reviewing! Added windows/linux fixes and tests passing 👍

**Pull request merged**